pkg/cluster/nodes/node.go

```diff
@@ -224,3 +225,33 @@ func (n *Node) WriteKubeConfig(dest string) error {
 
 	return ioutil.WriteFile(dest, buff.Bytes(), 0600)
 }
+
+// WaitForControlPlane uses kubectl inside the "node" container to check if
+// the control plane node status is "Ready". The control plane node is found by
+// looking for the node containing the substring "control-plane" in its name.
+func (n *Node) WaitForControlPlane(until time.Time) bool {
+	return tryUntil(until, func() bool {
+		cmd := n.Command("kubectl", "--kubeconfig=/etc/kubernetes/admin.conf", "get", "nodes")
```

Comment:
adding `"--selector=node-role.kubernetes.io/master"` should eliminate the need to filter below